runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7501,19 +7458,33 @@ static void handleOffHeapDataForArrays(
          fej9->getOffsetOfDiscontiguousDataAddrField(), fej9->getOffsetOfContiguousDataAddrField());
 
       TR::Register *discontiguousDataAddrOffsetReg = srm->findOrCreateScratchRegister();
-      generateRegRegInstruction(TR::InstOpCode::XOR4RegReg, node, discontiguousDataAddrOffsetReg, discontiguousDataAddrOffsetReg, cg);
+      generateRegRegInstruction(TR::InstOpCode::XORRegReg(), node, discontiguousDataAddrOffsetReg, discontiguousDataAddrOffsetReg, cg);
       generateRegImmInstruction(TR::InstOpCode::CMPRegImm4(), node, sizeReg, 1, cg);
       generateRegImmInstruction(TR::InstOpCode::ADCRegImm4(), node, discontiguousDataAddrOffsetReg, 0, cg);
+
       dataAddrMR = generateX86MemoryReference(targetReg, discontiguousDataAddrOffsetReg, 3, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg);
       dataAddrSlotMR = generateX86MemoryReference(targetReg, discontiguousDataAddrOffsetReg, 3, fej9->getOffsetOfContiguousDataAddrField(), cg);
+      // Load first data element address
+      generateRegMemInstruction(TR::InstOpCode::LEARegMem(), node, tempReg, dataAddrMR, cg);
+
+      // Clear out tempReg if dealing with 0 length array
+      zeroReg = srm->findOrCreateScratchRegister();
+      generateRegRegInstruction(TR::InstOpCode::XORRegReg(), node, zeroReg, zeroReg, cg);
```

Comment:
This can use `TR::InstOpCode::XOR4RegReg` instead.  On x86 a 32-bit operation will zero the upper 32 bits of a register running in 64-bit mode, so the `XOR` only needs to explicitly operate on the lower 32 bits.

Reply:
ah gotcha, didn't know that. I'll revert my change.